src/include/migraphx/literal.hpp

```diff
@@ -50,6 +50,8 @@ struct literal : raw_data<literal>
      */
     explicit literal(shape::type_t shape_type) : m_shape(shape_type, {}) {}
 
+    explicit literal(const shape& s) : m_shape(s) {}
```

Comment:
Why is this added? A `literal` should always have a buffer set when the shape is not empty.

Reply:
It was required to save/load `hip::copy_literal` as part of the test inside `test/operators.cpp`.  It wasn't linking any backend libraries so, it was only testing core MIGraphX Ops. As part of this PR, I made it to link `migraphx_all_targets` and it started testing all backend ops as wel. 

During serialization literals only generate `data` field if  it not empty but during de-serialization it expects "data"  field to be present and it would fail for `hip::copy_literal` save/load test since it has empty literal.